src/query/procedure/mg_procedure_impl.hpp

```diff
@@ -545,14 +545,38 @@ struct mgp_result {
   std::optional<utils::pmr::string> error_msg;
 };
 
+struct mgp_func_result {
+  explicit mgp_func_result(mgp_value *value, std::optional<utils::pmr::string> error_msg)
+      : value(value), error_msg(error_msg) {}
+
+  explicit mgp_func_result() {}
+  /// Return Magic function result with potential error
+  mgp_value *value;
+  std::optional<utils::pmr::string> error_msg;
+};
+
 struct mgp_graph {
   query::DbAccessor *impl;
   storage::View view;
   // TODO: Merge `mgp_graph` and `mgp_memory` into a single `mgp_context`. The
   // `ctx` field is out of place here.
   query::ExecutionContext *ctx;
+
+  static mgp_graph WritableGraph(query::DbAccessor &acc, storage::View view, query::ExecutionContext &ctx) {
+    return mgp_graph{&acc, view, &ctx};
+  }
+
+  static mgp_graph NonWritableGraph(query::DbAccessor &acc, storage::View view) {
+    return mgp_graph{&acc, view, nullptr};
+  }
 };
 
+struct mgp_func_context {
+  query::DbAccessor *impl;
+  storage::View view;
```

Comment:
I think we agreed on preventing the user from using the `mgp_graph` at all. Also I don't think that it would be clean to have a `mgp_graph`  wrapper for functions

Reply:
This can be solved by unified `mgp_context` structure.